### PR TITLE
Use VSTS clean steps, remove manual clean

### PR DESCRIPTION
**Pull request by chcosta**
Complete https://github.com/dotnet/core-eng/issues/2918

Remove manual cleanup steps from Linux definitions, use VSTS cleaning.

### CODE REVIEW

File: buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json [hunk 1]
```diff
@@ -499,10 +465,6 @@
     "VsoPassword": {
       "value": null,
       "isSecret": true
-    },
-    "AgentToolsPath": {
```
Comment (weshaggard) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: This is unrelated and should be kept. @JohnTortugo just added these recently.

Comment (weshaggard) [reply buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: Nevermind I just noticed that it just got moved and not removed.

Comment (chcosta) [reply buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: It just got moved.

Comment (chcosta) [reply buildpipeline/DotNet-CoreFx-Trusted-Linux-Crossbuild.json]: VSTS likes to sort them alphabetically and that seemed reasonable to me.
